### PR TITLE
[devcenter] Update control plane and data plane APIs to v2023-10-01-preview

### PR DESCRIPTION
**Pull request by am-lim**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
devcenter

File: src/devcenter/azext_devcenter/aaz/latest/devcenter/admin/network_connection/_update.py [hunk 1]
```diff
@@ -74,15 +78,12 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
-            help="Active Directory domain name.",
+            help="Active Directory domain name",
         )
         _args_schema.domain_password = AAZStrArg(
             options=["--domain-password"],
             arg_group="Properties",
-            help="The password for the account used to join domain.",
-            blank=AAZPromptInput(
-                msg="Domain password:",
-            ),
```
Comment (kairu-ms) [inline src/devcenter/azext_devcenter/aaz/latest/devcenter/admin/network_connection/_update.py:85]: Why removed the prompt input support?

Comment (am-lim) [reply src/devcenter/azext_devcenter/aaz/latest/devcenter/admin/network_connection/_update.py]: Unintended, re-added this and also updated the aaz PR

**Review by kairu-ms — APPROVED** [review]

**Comment by kairu-ms** [comment]
Is this PR ready to review?

**Comment by kairu-ms** [comment]
BTW, we will add support in aaz-dev-tools for data-plane command generation in the next version. You can try it next time.

**Comment by am-lim** [comment]
> Is this PR ready to release?

Hi @kairu-ms, added one last change. The PR is now ready and the AAZ PR as well

**Pull request merged**